cordapp-cpk/src/test/resources/external-cordapp/build.gradle

```diff
@@ -0,0 +1,61 @@
+plugins {
+    id 'net.corda.plugins.cordapp-cpk'
+}
+
+evaluationDependsOnChildren()
+
+allprojects { p ->
+    println(p.path)
+
+    if (p.path in [rootProject.path, ":corda-platform-cordapp", ":external-cordapp-transitive-dependency"]) {
+        apply plugin: 'java-library'
+        apply plugin: 'maven-publish'
+        apply from: "$rootDir/repositories.gradle"
+        apply from: "$rootDir/javaTarget.gradle"
+
+
+        group = 'com.example'
+        version = cordapp_version
+
+        publishing {
+            repositories {
+                maven {
+                    name = 'Test'
+                    url = getProperty("maven.repository.dir")
+                }
+            }
+            publications {
+                maven(MavenPublication) {
+                    from components.java
+                    pluginManager.withPlugin('net.corda.plugins.cordapp-cpk') {
+                        artifact tasks.named('cpk')
+                    }
+                }
+            }
+        }
+
+        jar {
+            archiveBaseName = p.name
+        }
+    }
+}
+
+cordapp {
+    targetPlatformVersion = platform_version.toInteger()
+    minimumPlatformVersion = platform_version.toInteger()
+
+    contract {
+        name = 'External CorDapp'
+        versionId = cordapp_contract_version.toInteger()
+        licence = 'Test-Licence'
+        vendor = 'R3'
+    }
+}
+
+dependencies {
+//    cordaProvided project(":corda-api")
```

Comment:
I do believe you can add this dependency back in, if you tweak the `allprojects {}` block above it. Perhaps just split the publishing out into a separate `allprojects {}` block?

Reply:
In fact, if you put the publishing into an `allproject {}` block that also applies the `java-library` and `maven-publish` plugins then you can probably get rid of `evaluationDependOnChildren()` too.